[jit] flip the "don't inline" switch

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26706 [jit] flip the "don't inline" switch**

This has been ready for some time, just waiting on services to push with
the new code.

Differential Revision: [D17543304](https://our.internmc.facebook.com/intern/diff/D17543304)

suo committed:
[jit] flip the "don't inline" switch

This has been ready for some time, just waiting on services to push with
the new code.

ZolotukhinM reviewed (approved):
Looks good to me!
What is our long-term plan regarding this? Is there a reason not to remove it completely eventually? It currently mixes "optimizations" and "post-IRgen" passes and I'd like to clean it up :)

suo commented:
Yep, we can remove. I'd prefer to leave the "switch-flipping" plumbing in for now just in case we need to flip it back in an emergency 😛

suo committed:
Update on "[jit] flip the "don't inline" switch"


This has been ready for some time, just waiting on services to push with
the new code.

Differential Revision: [D17543304](https://our.internmc.facebook.com/intern/diff/D17543304)

suo committed:
Update on "[jit] flip the "don't inline" switch"


This has been ready for some time, just waiting on services to push with
the new code.

Differential Revision: [D17543304](https://our.internmc.facebook.com/intern/diff/D17543304)
